src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_Commit.cs

```diff
@@ -52,9 +52,18 @@ private CompletionProvider GetCompletionProvider(CompletionItem item)
             this.StopModelComputation();
 
             CompletionChange completionChange;
-            using (var transaction = new CaretPreservingEditTransaction(
+            using (var transaction = CaretPreservingEditTransaction.TryCreate(
                 EditorFeaturesResources.IntelliSense, TextView, _undoHistoryRegistry, _editorOperationsFactoryService))
             {
+                if (transaction == null)
+                {
+                    // This text buffer has no undo history and has probably been unmapped.
```

Comment:
As discussed earlier, this prevents the crash tracked by internal bug 213801. That crash will be fixed (with presumably no work required on our side) by the editor after RC. This is a change to work around that crash.